autowrap/ConversionProvider.py

```diff
@@ -1152,12 +1151,16 @@ def output_conversion(
                 locals(),
             )
             return code
+        # Code for value AND key that is wrapped
         elif (
-            not cy_tt_key.is_enum
-            and tt_key.base_type in self.converters.names_of_wrapper_classes
+            not cy_tt_value.is_enum
+            and self.is_wrapper_class(tt_value.base_type)
+            and self.is_wrapper_class(py_tt_key.base_type)
         ):
-            value_conv = "<%s>(deref(%s).second)" % (cy_tt_value, it)
+            key_conv = "deref(<%s *> (<%s> key).inst.get())" % (cy_tt_key, py_tt_key)
+            cy_tt = tt_value.base_type
```

Comment:
cy_tt_value_base? Feel free to use better variable names. You do not need to stay with the old ones.